charon/pkgs/npm.py

```diff
@@ -86,48 +87,47 @@ def handle_npm_uploading(
 
         Returns the directory used for archive processing and if uploading is successful
     """
-    target_dir, valid_paths, package_metadata = _scan_metadata_paths_from_archive(
-        tarball_path, prod=product, dir__=dir_
-    )
-    if not os.path.isdir(target_dir):
-        logger.error("Error: the extracted target_dir path %s does not exist.", target_dir)
-        sys.exit(1)
+    for target in targets:
+        bucket_ = target[1]
+        prefix__ = remove_prefix(target[2], "/")
+        registry__ = target[3]
+        target_dir, valid_paths, package_metadata = _scan_metadata_paths_from_archive(
+            tarball_path, registry__, prod=product, dir__=dir_
+        )
+        if not os.path.isdir(target_dir):
+            logger.error("Error: the extracted target_dir path %s does not exist.", target_dir)
+            sys.exit(1)
+        valid_dirs = __get_path_tree(valid_paths, target_dir)
+
+        # main_target = targets[0]
+        client = S3Client(aws_profile=aws_profile, dry_run=dry_run)
```

Comment:
To save performance, we don't need to initiate this client multi times for different targets.

Reply:
@ligangty, fixed, thanks!